### PR TITLE
HPCC-25071 Add authorization feature for WsWorkunits.WUDeployWorkunit

### PR DESCRIPTION
**Pull request by wangkx**
A new authorization feature is added for WsWorkunits.WUDeployWorkunit.
Without this change, any user who has rights to create a workunit can
request the WsWorkunits.WUDeployWorkunit. With this fix, a user has to
apply for the rights to accessing this new feature before the user can
call this function.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: initfiles/componentfiles/configschema/xsd/esp_service_smc.xsd [hunk 1]
```diff
@@ -124,6 +124,9 @@
                                     <AuthenticateFeature description="Access to others&apos; workunits"
                                                          path="OthersWorkunitsAccess" resource="OthersWorkunitsAccess"
                                                          service="ws_workunits"/>
+                                    <AuthenticateFeature description="Access to deploy&apos; workunits"
```
Comment (afishbeck) [inline initfiles/componentfiles/configschema/xsd/esp_service_smc.xsd:127]: apos not needed here.

File: initfiles/componentfiles/configschema/xsd/buildset.xml [hunk 2]
```diff
@@ -67,6 +67,7 @@
           <AuthenticateFeature description="Access to cluster topology" path="ClusterTopologyAccess" resource="ClusterTopologyAccess" service="ws_topology"/>
           <AuthenticateFeature description="Access to own workunits" path="OwnWorkunitsAccess" resource="OwnWorkunitsAccess" service="ws_workunits"/>
           <AuthenticateFeature description="Access to others&apos; workunits" path="OthersWorkunitsAccess" resource="OthersWorkunitsAccess" service="ws_workunits"/>
+          <AuthenticateFeature description="Access to deploy&apos; workunits" path="DeployWorkunitsAccess" resource="DeployWorkunitsAccess" service="ws_workunits"/>
```
Comment (afishbeck) [inline initfiles/componentfiles/configschema/xsd/buildset.xml:70]: apos not needed here.

File: initfiles/componentfiles/configxml/buildsetCC.xml.in [hunk 3]
```diff
@@ -201,6 +201,10 @@
                         path="OthersWorkunitsAccess"
                         resource="OthersWorkunitsAccess"
                         service="ws_workunits"/>
+                    <AuthenticateFeature description="Access to deploy&apos; workunits"
```
Comment (afishbeck) [inline initfiles/componentfiles/configxml/buildsetCC.xml.in:204]: Same.

File: initfiles/componentfiles/configxml/buildsetCC.xml.in [hunk 4]
```diff
@@ -279,6 +283,10 @@
                         path="OthersWorkunitsAccess"
                         resource="OthersWorkunitsAccess"
                         service="ws_sql"/>
+                    <AuthenticateFeature description="Access to deploy&apos; workunits"
```
Comment (afishbeck) [inline initfiles/componentfiles/configxml/buildsetCC.xml.in:286]: same.

File: initfiles/etc/DIR_NAME/environment.xml.in [hunk 5]
```diff
@@ -476,6 +481,11 @@
                          path="OthersWorkunitsAccess"
                          resource="OthersWorkunitsAccess"
                          service="ws_sql"/>
+    <AuthenticateFeature authenticate="Yes"
+                         description="Access to deploy&apos; workunits"
```
Comment (afishbeck) [inline initfiles/etc/DIR_NAME/environment.xml.in:485]: Same.

**Review by afishbeck — CHANGES_REQUESTED** [review]
Looks good except for the extra apostrophes.

**Comment by afishbeck** [comment]
@wangkx go ahead and squash now and then I'll approve.

**Comment by wangkx** [comment]
@afishbeck squashed. I just created https://github.com/hpcc-systems/LN/pull/982. Please review.

File: initfiles/componentfiles/configschema/xsd/esp_service_smc.xsd [hunk 6]
```diff
@@ -121,9 +121,12 @@
                                                          service="ws_topology"/>
                                     <AuthenticateFeature description="Access to own workunits" path="OwnWorkunitsAccess"
                                                          resource="OwnWorkunitsAccess" service="ws_workunits"/>
-                                    <AuthenticateFeature description="Access to others&apos; workunits"
```
Comment (afishbeck) [inline initfiles/componentfiles/configschema/xsd/esp_service_smc.xsd:124]: Actually I should have looked before you squashed.  Here the apostrophe was correct.  It indicates that they own the workunits.

She has access to the others' houses.
She is allowed to deploy resources.

Comment (wangkx) [reply initfiles/componentfiles/configschema/xsd/esp_service_smc.xsd:124]: My mistake. The apostrophe is needed for others, not for deploy.

**Comment by wangkx** [comment]
@afishbeck The apos is added back for both PRs. I squashed so that that would be easier for review.

**Review by afishbeck — APPROVED** [review]
@wangkx looks good.

**Pull request merged**